src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -523,18 +524,15 @@ void D4NFilterObject::set_attrs_from_obj_state(const DoutPrefixProvider* dpp, op
   return;
 }
 
-int D4NFilterObject::calculate_version(const DoutPrefixProvider* dpp, optional_yield y, std::string& version)
+int D4NFilterObject::calculate_version(const DoutPrefixProvider* dpp, optional_yield y, std::string& version, rgw::sal::Attrs& attrs)
 {
   //versioned objects have instance set to versionId, and get_oid() returns oid containing instance, hence using id tag as version for non versioned objects only
   ldpp_dout(dpp, 10) << "D4NFilterObject::" << __func__ << "(): object name: " << this->get_name() << " instance: " << this->have_instance() << dendl;
   if (! this->have_instance() && version.empty()) {
-    bufferlist bl;
-    if (this->get_attr(RGW_ATTR_ID_TAG, bl)) {
-      version = bl.c_str();
+    bufferlist bl = attrs[RGW_ATTR_ID_TAG];
+    version = bl.to_str();
```

Comment:
the old code had version set to bl.c_str(), is there a reason for changing that to bl.to_str()?

Reply:
No, I will update this

Reply:
ok